src/quicr_client.cpp

```diff
@@ -227,6 +326,93 @@ QuicRClient::publishNamedObjectFragment(const quicr::Name& /* quicr_name */,
   throw std::runtime_error("UnImplemented");
 }
 
+bool
+QuicRClient::notify_pub_fragment(const messages::PublishDatagram& datagram,
+                                 const std::map<int, bytes>& frag_map)
+{
+  if ((frag_map.rbegin()->first & 0x1) != 0x1) {
+    return false; // Not complete, return false that this can NOT be deleted
+  }
+
+  bytes reassembled;
+
+  int seq_bytes = 0;
+  for (const auto& item : frag_map) {
+    if ((item.first >> 1) - seq_bytes != 0) {
+      // Gap in offsets, missing data, return false that this can NOT be deleted
+      return false;
+    }
+
+    reassembled.insert(
+      reassembled.end(), item.second.begin(), item.second.end());
+    seq_bytes += item.second.size();
+  }
+
+  for (const auto& entry : sub_delegates) {
+    quicr::bytes copy = reassembled;
+
+    if (entry.first.contains(datagram.header.name)) {
+      sub_delegates[entry.first]->onSubscribedObject(
+        datagram.header.name, 0x0, 0x0, false, std::move(copy));
+    } else {
+      std::cout << "Name:" << datagram.header.name.to_hex()
+                << ", not in namespace " << entry.first.to_hex() << std::endl;
+    }
+  }
+
+  return true;
+}
+
+void
+QuicRClient::handle_pub_fragment(messages::PublishDatagram&& datagram)
+{
+  static int cindex = 1;
```

Comment:
Should this be static? Are we guaranteed that by the end of the function cindex will be set to 1 again? Or do we not care?

Reply:
Yes, it's persistent and is maintained within this method.  It'll wrap back to 1 when reaching MAX FRAGMENT BUFFERS.  I suppose we can move it to the class var, although this is the only method that uses it.